### ISSUE TITLE
Supporting combinations of boot options (eg, graphical fallback mode + load into RAM + no energy saving)

### ISSUE THREAD

**Issue by shasheene** [rescuezilla/rescuezilla#177]
Rescuezilla v1.0.6 switched to the GRUB bootloader from ISOLINUX (SYSLINUX) to support booting Rescuezilla on EFI-only machines (including with Secure Boot enabled), in addition to legacy BIOS and CD-ROM. This was definitely a worthwhile switch, and GRUB is definitely the most popular bootloader for Linux systems. Unfortunately GRUB's design has some major limitations around the customization of the boot menu, and not just the lack of ability to [display help text when highlighting a GRUB menu item (task #76)](https://github.com/rescuezilla/rescuezilla/issues/76).

Increasingly a small number of users want to load Rescuezilla with certain combinations of boot options, such as "graphical fallback mode", "load USB into RAM" _and_ "Disable energy saving", **but the user has no way to select a combination of options** without manual typing to edit the boot options. SYSLINUX a solution for this: the ability to define an [advanced menu](https://wiki.syslinux.org/wiki/index.php?title=Menu#Introduction) which allows a keyboard's function keys (F1-F12) to select toggle options from a submenu. Consider the animated GIF below displaying the Ubuntu 20.04's ISOLINUX boot menu (which is used when the installation media is booted using a CD-ROM and a key is pressed during boot):

![Ubuntu 20.04 ISOLINUX](https://user-images.githubusercontent.com/1414379/105619041-6fb44300-5e3e-11eb-8f0d-1309b3afee46.gif)

The approach above is more powerful than Ubuntu 20.04's GRUB menu (displayed when booting on EFI systems), or Rescuezilla's similar (but themed) interface:

<a href="https://user-images.githubusercontent.com/1414379/105619155-d6862c00-5e3f-11eb-810a-4df0964e2e41.png"><img width=350 src="https://user-images.githubusercontent.com/1414379/105619155-d6862c00-5e3f-11eb-810a-4df0964e2e41.png"></a>  <a href="https://user-images.githubusercontent.com/1414379/105619253-82c81280-5e40-11eb-95a8-80328f733bb1.png"><img width=350 src="https://user-images.githubusercontent.com/1414379/105619253-82c81280-5e40-11eb-95a8-80328f733bb1.png"></a> <a href="https://user-images.githubusercontent.com/1414379/105619171-ee5db000-5e3f-11eb-9f8f-3b1d40529c4a.png"><img width=350 src="https://user-images.githubusercontent.com/1414379/105619171-ee5db000-5e3f-11eb-9f8f-3b1d40529c4a.png"></a>

The biggest downside to SYSLINUX's approach is it is overly complex for newbies. With Rescuezilla a user can easily pick the language they want (which automatically selects the keyboard layout) and off they go, and from the graphical environment they'll be able to [select a different keyboard layout](https://github.com/rescuezilla/rescuezilla/issues/91) using a simple application. More design work needs to be done because SYSLINUX's function key approach is definitely not simple enough for Rescuezilla.

**This task is a very low priority stretch goal**, but its important to flag it because it may become increasingly important. Rescuezilla is likely not the only Linux distribution that is affected by this. The solution of this task will likely involve forking the GRUB bootloader and redesigning its internals. While I'm at it I may as well add support [displaying help text when highlighting a GRUB menu item (task #76)](https://github.com/rescuezilla/rescuezilla/issues/76) can be done too). All work needs to be upstreamed to the GRUB bootloader project itself as Rescuezilla Canonical needs to sign the package for EFI Secure Boot to continue to work.

**Comment by GralfR** [comment]
The rescuezilla bootmenu is great as it is in 2.1.2! If I need a live-system I  want to get it booted as fast as possible. It's annoying to click thru all the options if you don't like the defaults. For a german user booting clonezilla for example takes a lot of time and concentration if you just connected a german keyboard. That is solved much better in rescuezilla!

But options to adjust the live-system at boot (e.g. disable ACPI #173) would be great. I think this additional menu should be an option in the current menu, like "advanced options". So one can just select language and boot up the system or can spend time in an advanced options menu. This advanced options menu could look like the clonezilla advanced men:
![image](https://user-images.githubusercontent.com/66326427/105626747-23fb9c80-5e32-11eb-8e63-97ef44288c07.png)
where each option can be selected by some kind of checkboxes or maybe dropdownlists, because the way the Ubuntu bootloader does it, will soon be out of F-keys if the number of options grow.

An additional option in the current rescuezilla boot menu could be "load settings from file" if there is an external settings-file like mentioned in #8 ([https://github.com/rescuezilla/rescuezilla/issues/8#issuecomment-766318771])

**Comment by shasheene** [comment]
Yeah OK. It sounds like the current approach is working pretty well for people.

Unfortunately GRUB doesn't have the ability to make checkboxes like you described. Any changes to the boot menu is a big task involving modifying the internals of the GRUB bootloader and getting the GRUB project to accept my changes. I think this is worthwhile for [displaying help text when highlighting a GRUB menu item (task #76)](https://github.com/rescuezilla/rescuezilla/issues/76), but the ability to select a combination of boot options will only be used by a small number expert users who can in the worst case manually edit the boot options themselves. In theory, because for some users their preferred keyboard won't be configured making eg, typing CTRL-X near impossible.

We'll figure out if task #173 can be achieved without a kernel boot option which would lessen the need for this task (at least until the next kernel boot option menu item).
